### PR TITLE
feat(home): add auth-aware sign-in

### LINKED ISSUE #132: Make homepage Sign-in link auth-aware (skip intermediate NextAuth page)
## User Request

The homepage "Sign in" link currently always navigates to `/api/auth/signin/logto?callbackUrl=/admin`, which shows an intermediate NextAuth sign-in page before redirecting to the Logto IdP. 

We want the homepage to detect auth status and:
- **Authenticated users:** Navigate directly to `/admin`
- **Unauthenticated users:** Go directly to Logto IdP (skip the intermediate page)

### Approach
Convert `src/app/page.tsx` to an **async server component** and call `getServerSession(authOptions)` to detect auth status. Extract the "Sign in" link into a small client component that uses `signIn("logto", { callbackUrl: "/admin" })` from `next-auth/react` for unauthenticated users (this bypasses the intermediate page by handling CSRF + POST automatically).

**Why server component?**
- No flash of wrong state (resolved before HTML is sent)
- Consistent with existing codebase pattern (`admin/layout.tsx`, etc.)
- No need for `SessionProvider` — `signIn()` from `next-auth/react` works standalone

#### 1. `src/app/page.tsx` (UPDATE)
- Make the default export `async`
- Import and call `getServerSession(authOptions)`
- Derive `isAuthenticated = !!session?.user?.id`
- Replace the hardcoded "Sign in" `<Link>` with a new `<SignInCta>` client component, passing `isAuthenticated`

#### 2. `src/components/sign-in-cta.tsx` (NEW)
Client component:
- **If authenticated:** Renders `<Link href="/admin">` with text like "Go to Admin" or "Sign in" (keeps current label)
- **If not authenticated:** Renders `<button onClick={() => signIn("logto", { callbackUrl: "/admin" })}>Sign in</button>`

How `signIn("logto", { callbackUrl: "/admin" })` works (next-auth v4):
1. Fetches `GET /api/auth/csrf` → gets CSRF token
2. POSTs to `/api/auth/signin/logto` with CSRF token + callbackUrl
3. next-auth generates PKCE challenge, state, nonce
4. Returns the Logto authorize URL
5. Browser redirects directly to Logto IdP — **no intermediate page**

#### 3. `tests/e2e/signin-link.spec.ts` (UPDATE)
- The "Sign in" element for unauthenticated users is now a `<button>`, not a `<Link>`
- Update E2E selectors accordingly
- Note: The two "Get Started" links (`#quick-start`) remain unchanged

#### 4. `src/components/__tests__/sign-in-cta.test.tsx` (NEW)
Unit tests for the `SignInCta` component:
- Authenticated: renders link to `/admin`
- Not authenticated: renders button
- Not authenticated + click: calls `signIn("logto", { callbackUrl: "/admin" })`

### Edge Cases
| Edge Case | Handling |
|---|---|
| Expired session | `getServerSession` returns `null` → unauthenticated flow |
| Session exists but user deleted | `session?.user?.id` falsy → unauthenticated flow |
| JS disabled | `<button>` won't work; acceptable since page is already JS-heavy. Optional: add `<noscript>` fallback |
| Race: session expires between SSR and click | User clicks → goes to `/admin` → admin layout redirects to sign-in. Correct existing behavior. |

### What Is NOT Changed
- `authOptions.pages.signIn` — not changed; intermediate page still exists for admin-layout redirects
- Root layout — no `SessionProvider` needed
- No new dependencies — `signIn` from `next-auth/react` is already available

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- make the homepage CTA auth-aware using the server session
- add a reusable SignInCta client component with unit coverage
- update landing page e2e expectations for the button CTA

## Testing
- pnpm lint
- pnpm typecheck
- pnpm test

Closes #132

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Summary
- make the homepage CTA auth-aware using the server session
- add a SignInCta client component with unit coverage
- update landing page e2e expectations for the button CTA

## Testing
- pnpm lint (no errors)
- pnpm typecheck (no errors)
- pnpm test (171 passed, 0 failed, 0 skipped)

File: src/components/sign-in-cta.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+type SignInCtaProps = {
+  isAuthenticated: boolean;
+  className?: string;
+  children?: ReactNode;
+};
+
+export function SignInCta({ isAuthenticated, className, children }: SignInCtaProps) {
+  const label = children ?? "Sign in";
+
```
Comment (noa-lucent) [inline src/components/sign-in-cta.tsx]: [major] `children ?? "Sign in"` is a "just-in-case" fallback that is never exercised — every call site passes explicit children (`"Get Started"`). This default is also misleading: the component is a generic CTA, yet the fallback hard-codes "Sign in".

Make `children` required (drop the `?`) and remove the fallback. If no children are passed, the caller should be explicit — not silently get a label that may be wrong.

```tsx
type SignInCtaProps = {
  isAuthenticated: boolean;
  className?: string;
  children: ReactNode;
};

export function SignInCta({ isAuthenticated, className, children }: SignInCtaProps) {
  // use `children` directly, no fallback
```

**Review by noa-lucent — CHANGES_REQUESTED** [review]
The approach is sound — server-side session check, clean client component split, proper test coverage. Nice work overall.

Two major items to address before merge:

1. **Remove `src/components/utils.ts`** — unnecessary re-export that breaks the codebase's consistent `@/lib/utils` import pattern.
2. **Make `children` required in `SignInCta`** — the `"Sign in"` fallback is a never-used "just-in-case" default that is also semantically misleading.

One minor: drop the `cn()` wrapper when passing a single class string.
